src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -303,6 +332,15 @@ class TestRequestRateManager : public TestLoadManagerBase,
     CheckSequences(params_.num_of_sequences);
   }
 
+  /// Test that the shared memory methods are called correctly
+  ///
+  void TestSharedMemory(uint request_rate, uint duration_ms)
+  {
+    ChangeRequestRate(request_rate);
```

Comment:
Would it be sufficient to design the test case such that it only calls the `Infer()` function once and then you go check that the memory stats mocks are called the appropriate number of times?

Reply:
I was simply modeling my test after one I knew was already working. 
It is possible we can refine this later